File: Aula04/Go/Atv04/main.go
package main

import (
	"fmt"
)

type Motor struct {
	Tipo     string
	Potencia int
}

type Pneu struct {
	Marca   string
	Tamanho int
}

type Carro struct {
	Marca  string
	Modelo string
	Motor  Motor
	Pneus  [4]Pneu
}

func (Motor *Motor) Ligar() {
	fmt.Println("O motor está ligado.")
}

func (Motor *Motor) Desligar() {
	fmt.Println("O motor está desligando.")
}

func (Pneu *Pneu) Inflar() {
	fmt.Println("O pneu está inflado.")
}

func (Pneu *Pneu) Desinflar() {
	fmt.Println("O pneu está desinflado.")
}

func (Carro *Carro) Ligar() {
	fmt.Println("O carro está pronto para rodar.")
}

func (Carro *Carro) Desligar() {
	fmt.Println("O carro foi desligado.")
}

func (Carro *Carro) Trocar_pneu(indice int, novo_pneu Pneu) {
	Carro.Pneus[indice] = novo_pneu
	fmt.Println("Pneu", (indice + 1), "trocado.")
}

func main() {

	pneus := [4]Pneu{{"Pirelli", 18}, {"Pirelli", 18}, {"Pirelli", 18}, {"Pirelli", 18}}

	carro := Carro{"Toyota", "Corolla", Motor{"Gasolina", 150}, pneus}
	carro.Ligar()

	novo_pneu := Pneu{"Michelin", 17}
	carro.Trocar_pneu(2, novo_pneu)
	carro.Desligar()

}

File: Aula04/Python/Atv04/main.py
class Motor:
    def __init__(self, tipo, potencia):
        self.tipo = tipo
        self.potencia = potencia
    
    def ligar(self):
        print("O motor está ligado.")

    def desligar(self):
        print("O motor está desligado.")

class Pneu:
    def __init__(self, marca, tamanho):
        self.marca = marca
        self.tamanho = tamanho
    
    def inflar(self):
        print("O pneu está inflado.")

    def desinflar(self):
        print("O pneu está desinflado.")

class Carro:
    def __init__(self, marca, modelo):
        self.marca = marca
        self.modelo = modelo
        self.motor = Motor("Gasolina", 150)
        self.pneus = [Pneu("Pirelli", 18) for _ in range(4)]

    def ligar(self):
        self.motor.ligar()
        print("O carro está pronto para rodar.")

    def desligar(self):
        self.motor.desligar()
        print("O carro foi desligado.")

    def trocar_pneu(self, indice, novo_pneu):
        self.pneus[indice] = novo_pneu
        print(f"Pneu {indice + 1} trocado.")

carro = Carro("Toyota", "Corolla")
carro.ligar()

novo_pneu = Pneu("Michelin", 17)
carro.trocar_pneu(2, novo_pneu)
carro.desligar()